packages/ui/src/components/va-virtual-scroller/VaVirtualScroller.vue

```diff
@@ -0,0 +1,126 @@
+<template>
+  <div ref="wrapper" class="va-virtual-scroller">
+    <div class="va-virtual-scroller__container" :style="containerStyleComputed">
+      <div
+        ref="list"
+        role="list"
+        class="va-virtual-scroller__list"
+        :class="listClassComputed"
+        :style="listStyleComputed"
+      >
+        <template
+          v-for="(item, index) in renderBuffer"
+          :key="uniqueKey(item, index)"
+        >
+            <slot v-bind="{item, index: renderStartIndex + index}" />
+        </template>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent, ref, computed } from 'vue'
+import pick from 'lodash/pick.js'
+import isArray from 'lodash/isArray.js'
+
+import { useEvent, useBem } from '../../composables'
+import { useVirtualScrollerSizes, useVirtualScrollerSizesProps } from './useVirtualScrollerSizes'
+
+import { warn } from '../../services/utils'
+
+export default defineComponent({
+  name: 'VaVirtualScroller',
+
+  props: {
+    ...useVirtualScrollerSizesProps,
+    items: { type: Array, required: true, validator: (v: any[]) => !!v.length },
```

Comment:
Items could definitely be an empty array.

Maybe we can also integrate some empty list handling, but up to you.

Reply:
Will discuss tomorrow - can't imagine the purpose of this case.